sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala

```diff
@@ -76,22 +85,51 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     }
   }
 
-  private def checkHiveQl(hiveQl: String): Unit = {
-    val df = sql(hiveQl)
+  // Used for generating new query answer files by saving
+  private val saveQuery = false
+
+  /**
+   * Compare the generated SQL with the expected answer string.
+   * Note that there exists a normalization for both arguments for the convenience.
+   * - Remove the id from the generated attributes, e.g., `gen_attr_1` -> `gen_attr`.
+   */
+  private def checkSQLStructure(originalSQL: String, convertedSQL: String, answerFile: String) = {
+    val normalizedGenSQL = convertedSQL.replaceAll("`gen_attr_\\d+`", "`gen_attr`")
+    if (answerFile != null) {
+      val path = s"src/test/resources/sqlgen/$answerFile.sql"
```

Comment:
Also instead of using a variable as a flag, maybe we can use an environmental variable, and then document in the beginning

```
* To re-generate golden files, run:  SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "hive/test-Only * LogicalPlanToSQLSuite"
```

something like this?